### PR TITLE
SR 75 Revised Clean PR

### PR DESCRIPTION
**Pull request by DmitriyMalayev**
# Task

installed and configured jest via npm init jest@latest.
options chose were typescript, babel and enabled code coverage

Created separate folder for jest testing, configured only that folder to run upon executing npx jest

Added dependencies (jest, testing library)
Updated babel config, and jest config

[Link to Ticket](Insert Link Here)

[Add links to additional documentation if exists. Otherwise, delete this block]

[Insert related tickets here. Otherwise, delete this block]

# Author Tasks

## Acceptance Criteria
Please include the acceptance criteria from the ticket here or select N/A.
- [ ] N/A

### CODE REVIEW

File: frontend/coverage/clover.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mcverter) [inline frontend/coverage/clover.xml:1]: per eugene's instruction, put the `coverage/` folder in `.gitignore`.  we don't want this to get committed to the repo

https://github.com/Incredihire/Survey/blob/main/.gitignore

File: frontend/jest.config.ts [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+  // testEnvironment: 'node',  
+  // Options that will be passed to the testEnvironment
+  // testEnvironmentOptions: {},
+
+  // Adds a location field to test results
+  // testLocationInResults: false,
+
+  // The glob patterns Jest uses to detect test files
+  testMatch: [
+    "<rootDir>/__tests__/**/*.test.ts"
```
Comment (mcverter) [inline frontend/jest.config.ts:159]: great.  this looks like the right way to specify the directory where tests are run.

File: frontend/jest.config.ts [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    // "**/__tests__/**/*.[jt]s?(x)",
+    // "**/?(*.)+(spec|test).[tj]s?(x)"
```
Comment (mcverter) [inline frontend/jest.config.ts:161]: I believe the playwright tests also use *.spec as a file pattern so I would not reuse it for Jest.  Maybe get rid of this line entirely

File: frontend/package.json [hunk 4]
```diff
@@ -8,7 +8,8 @@
     "build": "tsc && vite build",
     "lint": "biome check --apply-unsafe --no-errors-on-unmatched --files-ignore-unknown=true ./",
     "preview": "vite preview",
-    "generate-client": "openapi-ts --input ./openapi.json --output ./src/client --client axios --exportSchemas true && biome format --write ./src/client"
+    "generate-client": "openapi-ts --input ./openapi.json --output ./src/client --client axios --exportSchemas true && biome format --write ./src/client",
```
Comment (mcverter) [inline frontend/package.json:11]: what is this?

Comment (DmitriyMalayev) [reply frontend/package.json:11]: this automate the process of generating a client library from an OpenAPI specification, it also automatically formats it adhering to guidelines

File: frontend/jest.config.ts (hunk 3 above)
Comment (DmitriyMalayev) [reply frontend/jest.config.ts:161]: That line is required. If I remove it it will run both tests

File: frontend/coverage/clover.xml (hunk 1 above)
Comment (DmitriyMalayev) [reply frontend/coverage/clover.xml:1]: resolved

**Review by ilhanbae — APPROVED** [review]

**Review by mcverter — APPROVED** [review]

**Pull request merged**